CRM/Dedupe/BAO/RuleGroup.php

```diff
@@ -233,7 +236,7 @@ public function fillTable() {
           if ($searchWithinDupes) {
             // get prepared to search within already found dupes if $searchWithinDupes flag is set
             $dao->query("DROP TEMPORARY TABLE IF EXISTS dedupe_copy");
-            $dao->query("CREATE TEMPORARY TABLE dedupe_copy SELECT * FROM dedupe WHERE weight >= {$weightSum}");
+            $dao->query("CREATE TEMPORARY TABLE dedupe_copy SELECT * FROM {$this->temporaryTables['dedupe']} WHERE weight >= {$weightSum}");
```

Comment:
This could use TempTable::createWithQuery().

Reply:
Yeah - I agree - I haven't included here because that will require a new chunk of digging into ensuring that test lines are passing through that code